File: src/main/java/br/uniflix/api/model/Filme.java
package br.uniflix.api.model;


import lombok.Getter;
import lombok.Setter;

import javax.persistence.*;
import javax.validation.constraints.*;
import java.util.List;

@Getter
@Setter
@Entity
public class Filme {

    @Id
    @GeneratedValue
    private Integer id;

    @NotEmpty
    @Size(max = 100, message = "O título não pode ter mais que 100 caracteres")
    private String titulo;

    @NotEmpty
    @Size(max = 1000, message = "A sinopse não pode ter mais que 1000 caracteres")
    private String sinopse;

    private String urlTrailer;

    @NotEmpty
    private String urlCapa;

    @Positive
    private Integer anoLancamento;

    @OneToMany(cascade = CascadeType.PERSIST)
    private List<Ator> elenco;

    @ManyToOne(cascade = CascadeType.PERSIST)
    private Genero genero;

    @OneToOne(cascade = CascadeType.PERSIST)
    private Video video;

}

File: uniflix-app/src/app/model/filme.ts
import { Genero } from './genero';
import { Ator } from './ator';
import { Video } from './video';
export interface Filme {
  id: number;
  titulo: string;
  sinopse: string;
  urlTrailer: string;
  urlCapa: string;
  anoLancamento: number;
  elenco: Ator[];
  genero: Genero;
  video: Video;
}
